app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_parser.rb

```diff
@@ -0,0 +1,44 @@
+module ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::EventParser
+  def self.event_to_hash(event, ems_id)
+    event_hash = {
+      :event_type => "#{event[:resource]}.#{event[:action]}",
+      :source     => "IBMCloud-PowerVS",
+      :ems_id     => ems_id,
+      :ems_ref    => event[:eventID],
+      :timestamp  => event[:time],
+      :full_data  => event,
+      :username   => event[:user][:email]
+    }
+
+    _log.debug("event_hash=#{event_hash}")
+    case event_hash[:event_type]
+    when /^pvm-instance/
+      parse_vm_event!(event, event_hash)
+    end
+
+    event_hash
+  end
+
+  def self.parse_vm_event!(event, event_hash)
+    pvm_instance_name = event[:message].split('\'')[1]
+    ems = ExtManagementSystem.find(event_hash[:ems_id])
+    vm = ems.vms.find_by(:name => pvm_instance_name)
+
+    if vm.nil? && ['update', 'create'].include?(event['type'])
+
+      # TODO: If event type is 'pvm-instance.update' and the VM name was
+      #       changed then only the new name is given in the event message :(
+      #       Should we refresh the entire EMS on this event? Same question
+      #       for a 'pvm-instance.create'.
+      event_hash
+    else
+      return
+    end
+
+    sleep(60)
```

Comment:
I hate to add calls to `sleep`... but the PowerVS back-end can be kind of slow. When testing MIQ would usually refresh the VM before the state had changed. I'm open to suggestions for a more elegant solution.

Reply:
Interesting so they raise an event about a state change before a `GET` would return the new state?  Is this like `pvm.instance.start` event and the state goes from stopped to starting but not running yet, or does it still return stopped?

Reply:
> Interesting so they raise an event about a state change before a `GET` would return the new state?

Correct. The event seems to be raised as soon as a state change request is made.

Here's an example using cli tools:
```
rdr-miq-mon01 > pcloud GET /pcloud/v1/cloud-instances/${cloud_instance_id}/events 'from_time=1638199159' | jq .
{
  "events": []
}
rdr-miq-mon01 > ic pi in c2c2a435-2cb9-4cfd-b5b3-3345dc01201b --json | jq .status                              
"ACTIVE"
rdr-miq-mon01 > ic pi instance-stop c2c2a435-2cb9-4cfd-b5b3-3345dc01201b         
Performing action stop for instance c2c2a435-2cb9-4cfd-b5b3-3345dc01201b under account IBM Power Cloud Development as user jwcarman@us.ibm.com...
OK
Action stop complete for instance c2c2a435-2cb9-4cfd-b5b3-3345dc01201b.
rdr-miq-mon01 > pcloud GET /pcloud/v1/cloud-instances/${cloud_instance_id}/events 'from_time=1638199159' | jq .
{
  "events": [
    {
      "action": "stop",
      "eventID": "80bf0d39-0ab5-4e30-af2c-9a18d33bbc46",
      "level": "info",
      "message": "Power virtual server instance 'rdr-rhacm-build-newname' stop action successfully completed.",
      "metadata": {},
      "resource": "pvm-instance",
      "time": "2021-11-29T15:20:52.777Z",
      "timestamp": 1638199252,
      "user": {
        "email": "jwcarman@us.ibm.com",
        "name": "Unknown Name",
        "userID": "IBMid-2700063YVN"
      }
    }
  ]
}
rdr-miq-mon01 > ic pi in c2c2a435-2cb9-4cfd-b5b3-3345dc01201b --json | jq .status                   
"ACTIVE"
rdr-miq-mon01 > sleep 60 && ic pi in c2c2a435-2cb9-4cfd-b5b3-3345dc01201b --json | jq .status
"SHUTOFF"
```

Reply:
That is unfortunate, really don't want to slow down the event catcher at all.  If we have to sleep somewhere I think this isn't the place, if someone powers on 10 vms in short order it'll be an 10 minutes before the last one is even queued for the event handler.

Reply:
I think we can use the sleep in the `EventCatcher::Stream` to force a delay (and allow it to be see via settings): f1bda4d